src/main/java/io/r2dbc/postgresql/client/ReactorNettyClient.java

```diff
@@ -195,9 +212,10 @@ private ReactorNettyClient(Connection connection) {
      * @param host               the host to connect to
      * @param port               the port to connect to
      * @param connectTimeout     connect timeout
+     * @param sslProvider        ssl provider for ssl protected connection
      * @throws IllegalArgumentException if {@code host} is {@code null}
      */
-    public static Mono<ReactorNettyClient> connect(ConnectionProvider connectionProvider, String host, int port, @Nullable Duration connectTimeout) {
+    public static Mono<ReactorNettyClient> connect(ConnectionProvider connectionProvider, String host, int port, @Nullable SslProvider sslProvider, @Nullable Duration connectTimeout) {
```

Comment:
Please encapsulate SSL config in a value object. Right now, the presence/absence of `SslProvider` makes some assumptions over SSL usage: Present: Require SSL. Absent SSL not supported. There are a lot of other options (see https://github.com/pgjdbc/pgjdbc/blob/master/docs/documentation/head/ssl-client.md) that we could want to support. And TBCH, having a non-nullable parameter is the better way to go.

Reply:
If user configured SslContext then ssl requirement is a pretty good assumption. And other options are about configuring sslcontext. We should provide either them (with all that cert, key, password, etc params) and build sslcontext on our own or SslContext param.

Reply:
Configuring `SslContext` isn't the most trivial thing to do. The API should support users in their intent to configure SSL in a convenient way.